packages/generator-common/src/compiler.ts

```diff
@@ -50,8 +53,39 @@ export async function transpileDirectory(
     allFiles2.map(file => resolve(path, file)),
     compilerOptions
   );
-  const emitResult = await program.emit();
 
+  // The write file handler does not support async function hence the work around with the outer promise list.
+  const fileWriterPromises: Promise<void>[] = [];
+  const prettierWriter: WriteFileCallback = (
+    fileName,
+    text,
+    writeByteOrderMark,
+    onError?,
+    sourceFiles?,
+    data?
+  ) => {
+    const parsed = parse(fileName);
+    const promise = mkdir(parsed.dir, { recursive: true }).then(() => {
+      // All not emitted files like .md or .json should be already prettified.
```

Comment:
[nth] General: In the section above we mention prettying, linting and formatting - I would prefer to reuse one word.